libraries/cmd/unit_generic.cpp

```diff
@@ -1664,13 +1664,18 @@ bool Unit::Explode(bool drawit, float timeit) {
             FactionUtil::GetRandExplosionAnimation(this->faction, bleh);
         }
         if (bleh.empty()) {
-            static Animation cache(game_options()->explosion_animation.c_str(), false, .1, BILINEAR, false);
+            static Animation * cache;
```

Comment:
I'd drop this code and see if the game still works. 
This whole philosophy of a resilient engine in the face of deficient assets is silly.
If we force Unit.pImage.explosion_type to have a valid value every time, we can drop all of this code:

```
        if (bleh.empty()) {
            FactionUtil::GetRandExplosionAnimation(this->faction, bleh);
        }
        if (bleh.empty()) {
            static Animation cache(game_options()->explosion_animation.c_str(), false, .1, BILINEAR, false);
            bleh = getRandomCachedAniString();
            if (bleh.size() == 0) {
                bleh = game_options()->explosion_animation;
            }
        }
```